db/migrate/20190716173854_add_secure_to_uploads.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class AddSecureToUploads < ActiveRecord::Migration[5.2]
+  def up
+    add_column :uploads, :secure, :boolean, default: false, null: false
+
+    if SiteSetting.prevent_anons_from_downloading_files
```

Comment:
We should not use models in migrations in case we ever rename of model, this would break the migrations.

Also, that loop should ideally be replace by 1 SQL query if possible.

Reply:
I don't know how to get the upload's `for_theme` and  `for_site_setting` attributes without loading the model. We don't want to mark these uploads as secure, they might be font files, for example.

Reply:
Except that any of [these `for_<something>` `attr_accessor`](https://github.com/discourse/discourse/blob/836b3f4d82dfc610574db93cfd1d86e4064d4062/app/models/upload.rb#L21-L26) are **not** stored in the database.

They're only used when creating an upload to do some different validation/processing depending on what the upload is for.

If you want to know if an upload is associated to a `theme` or a `site_setting`, you'll have to `join` some tables. Have a look at the [`clean_up_uploads`](https://github.com/discourse/discourse/blob/master/app/jobs/scheduled/clean_up_uploads.rb) job.